### PR TITLE
[AF-527] Support location objects in manifest

### PR DESCRIPTION
**Pull request by adammw**
:v: 

/cc @zendesk/vegemite

⛔️ **depends on https://github.com/zendesk/zendesk_app_framework/pull/657** ⛔️ 

### Description
Support location objects in manifest
e.g. 
```
{
  "location": {
    "support": {
      "background": {
        "url": "assets/iframe.html"
      },
      "nav_bar": {
        "url": "assets/iframe.html",
        "autoHide": true
      },
      "ticket_sidebar": {
        "url": "assets/iframe.html",
        "autoLoad": false
      },
      "new_ticket_sidebar": {
        "autoLoad": false
      }
    }
}
```

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-527

### Risks
* medium: generates source with a different location format

### CODE REVIEW

File: lib/zendesk_apps_support/validations/manifest.rb [hunk 1]
```diff
@@ -156,8 +156,15 @@ def invalid_location_error(package)
                                   count: invalid_locations.length)
             end
 
-            locations.values.each do |path|
-              errors << invalid_location_uri_error(package, path)
+            locations.each do |location_key, location|
+              url = location['url']
+              no_iframe = location.fetch('noIframe', false)
+              if url && !url.empty?
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/manifest.rb]: why the empty check?

**Comment by sandlerr** [comment]
now that this is an object rather than a string, is it a good time to stop using the `_legacy` hack?

**Comment by adammw** [comment]
😅  i'd love to (i.e. some sort of "legacy": true thing), but we're going to need to support _legacy for some time anyway since it gets serialised into the app source. one step at a time :)

**Comment by sandlerr** [comment]
right, thought now would be a good time to introduce a new way

File: lib/zendesk_apps_support/validations/manifest.rb [hunk 2]
```diff
@@ -156,8 +156,14 @@ def invalid_location_error(package)
                                   count: invalid_locations.length)
             end
 
-            locations.values.each do |path|
-              errors << invalid_location_uri_error(package, path)
+            locations.each do |location_key, location|
+              url = location['url']
+              auto_load = location.fetch('autoLoad', true)
+              if url && !url.empty?
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/manifest.rb]: why the empty check? we should already be checking valid url somewhere?

**Comment by adammw** [comment]
Closing as #144 has been merged

**Pull request closed without merging**